esmvaltool/diag_scripts/hydrology/lisflood.py

```diff
@@ -0,0 +1,143 @@
+"""LISFLOOD diagnostic."""
+import logging
+from pathlib import Path
+
+import iris
+from iris.analysis.maths import exp as iris_exp
+
+from esmvaltool.diag_scripts.shared import (ProvenanceLogger,
+                                            get_diagnostic_filename,
+                                            group_metadata, run_diagnostic)
+
+logger = logging.getLogger(Path(__file__).name)
+
+
+def get_provenance_record(ancestor_files):
+    """Create a provenance record."""
+    record = {
+        'caption': "Forcings for the LISFLOOD hydrological model.",
+        'domains': ['global'],
+        'authors': [
+            'verhoeven_stefan',
+            'kalverla_peter',
+        ],
+        'projects': [
+            'ewatercycle',
+        ],
+        'references': [
+            'acknow_project',
+        ],
+        'ancestors': ancestor_files,
+    }
+    return record
+
+
+def get_input_cubes(metadata):
+    """Create a dict with all (preprocessed) input files."""
+    inputs = {}
+    ancestors = {}
+    for attributes in metadata:
+        short_name = attributes['short_name']
+        if short_name in inputs:
+            raise ValueError(f"Multiple input files found for variable '{short_name}'.")
+        filename = attributes['filename']
+        logger.info("Loading variable %s", short_name)
+        cube = iris.load_cube(filename)
+        cube.attributes.clear()
+        inputs[short_name] = cube
+        ancestors[short_name] = [filename]
+
+    return inputs, ancestors
+
+
+def shift_era5_time_coordinate(cube, shift=30):
+    """Shift instantaneous variables (default = 30 minutes forward in time).
+
+    After this shift, as an example:
+    time format [1990, 1, 1, 11, 30, 0] will be [1990, 1, 1, 12, 0, 0].
+    For aggregated variables, already time format is [1990, 1, 1, 12, 0, 0].
+    """
+    time = cube.coord(axis='T')
+    time.points = time.points + shift / (24 * 60)
+    time.bounds = None
+    time.guess_bounds()
+    return cube
+
+
+def compute_vapour_pressure(d2m):
+    """Compute vapour pressure using tetens formula."""
+    # taken from Eq. 3.21 of Goudriaan (1977; https://library.wur.nl/WebQuery/wurpubs/70980)
+    e0 = 6.10588
```

Comment:
@sverhoeven I think it'd be good to add a check here that d2m is in Celcius (or Kelvin, already forgot what it needs to be). Since people might be tempted to change this in the recipe without noting that this affects the calculation here.